stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java

```diff
@@ -0,0 +1,136 @@
+package alluxio.stress.cli;
+
+import alluxio.AlluxioURI;
+import alluxio.ClientContext;
+import alluxio.client.file.FileSystemContext;
+import alluxio.client.file.FileSystemMasterClient;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.status.AlluxioStatusException;
+import alluxio.grpc.CreateDirectoryPOptions;
+import alluxio.grpc.CreateFilePOptions;
+import alluxio.grpc.DeletePOptions;
+import alluxio.grpc.SetAttributePOptions;
+import alluxio.master.MasterClientContext;
+import alluxio.resource.CloseableResource;
+import alluxio.stress.PinListFileSystemMasterClient;
+import alluxio.stress.rpc.GetPinnedFileIdsParameters;
+import alluxio.stress.rpc.RpcTaskResult;
+import alluxio.util.CommonUtils;
+import alluxio.util.FormatUtils;
+import com.beust.jcommander.ParametersDelegate;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class GetPinnedFileIdsBench extends RpcBench<GetPinnedFileIdsParameters> {
+  private static final Logger LOG = LoggerFactory.getLogger(GetPinnedFileIdsBench.class);
+
+  @ParametersDelegate
+  private GetPinnedFileIdsParameters mParameters = new GetPinnedFileIdsParameters();
+
+  private final InstancedConfiguration mConf = InstancedConfiguration.defaults();
+  private final FileSystemContext mFileSystemContext;
+  private final AlluxioURI mBaseUri;
+
+  {
+    mFileSystemContext = FileSystemContext.create(mConf);
+    mBaseUri = new AlluxioURI(mParameters.mBasePath);
+  }
+
+  @Override
+  public void prepare() throws Exception {
+    try (CloseableResource<alluxio.client.file.FileSystemMasterClient> client =
+             mFileSystemContext.acquireMasterClientResource()) {
+      client.get().createDirectory(mBaseUri,
+              CreateDirectoryPOptions.newBuilder().setAllowExists(true).build());
+    }
+    int numFiles = mParameters.mNumFiles;
+    int fileNameLength = (int) Math.max(8, Math.log10(numFiles));
+
+    CompletableFuture<Void>[] futures = new CompletableFuture[numFiles];
+    LOG.info("Generating {} test files", numFiles);
+    for (int i = 0; i < numFiles; i++) {
```

Comment:
Is this a sequential step to create all pinned files?
How long does it take to run for creating X files?

Reply:
The `prepare()` step is not included in timing so this will not affect the result precision, per se. It does take much time generating a big number of files, but based on the current API we don't have a good way to avoid that. Based on my experience running these tests, it takes a few minutes to generate a few million of files/blocks, on a 4-core/8-core master machine. So I would say the usability is okay for the current scale in testing. 

If we think of a billion files/blocks, that will probably take a beefy machine (24-core and up) and probably 1hr to prepare.